demos-europe/demosplan-core
# fix(refs DPLAN-12614): isNotEmptyFeatureCollection function returns false always

### Ticket
https://demoseurope.youtrack.cloud/issue/DPLAN-12614/Nach-dem-Geltungsbereich-angezeichnet-ist-bleibt-der-Titel-mit-Lage-einzeichnen


This function appears to be outdated since `this.territory.features` is not an array but an object so the `JSON.parse(string).features.length > 0` comparison could never work properly because it always returned false. To solve this we need to iterate over the keys of the features and check if the length is bigger than 0.

Also if the `isNotEmptyFeatureCollection` encounters errors, I think the correct way to handle this is to return false instead so the user gets the information that no territory has been set up so far. Also I think we should return the error in a `console.error()`

### How to review/test
- checkout branch
- login as FPA (e.g.)
- select any procedure
- navigate to "Antragsunterlagen" via the side menu
- add a territory
- save
- make sure if there is a territory set that the link appears in green color otherwise it should indicate there is no territory by rendering a default text color (blueish)

### PR Checklist


Delete the checkbox if it doesn't apply/isn't necessary.
